redfish/provider/resource_redfish_power.go

```diff
@@ -136,10 +142,10 @@ func (r *powerResource) Create(ctx context.Context, req resource.CreateRequest,
 		return
 	}
 	// 	// Lock the mutex to avoid race conditions with other resources
-	redfishMutexKV.Lock(plan.RedfishServer.Endpoint.ValueString())
-	defer redfishMutexKV.Unlock(plan.RedfishServer.Endpoint.ValueString())
+	redfishMutexKV.Lock(plan.RedfishServer[0].Endpoint.ValueString())
+	defer redfishMutexKV.Unlock(plan.RedfishServer[0].Endpoint.ValueString())
 
-	service, err := NewConfig(r.p, &plan.RedfishServer)
+	service, err := NewConfig(r.p, &plan.RedfishServer[0])
```

Comment:
Can we push the [0] logic into the NewConfig function? Then we wont need to make changes in all ds and res

Reply:
Better yet, let's push the locking logic as well to NewConfig.

Reply:
pushed the [0] logic to NewConfig Function